src/ui/components/select-dropdown/test/index.js

```diff
@@ -39,15 +39,16 @@ describe( 'index', () => {
 			expect( dropdown.find( '.select-dropdown.is-open' ).length ).toBe( 0 );
 		} );
 
-		test( 'should execute toggleDropdown when clicked', () => {
-			const originalToggleDropdown = SelectDropdown.prototype.toggleDropdown;
-			SelectDropdown.prototype.toggleDropdown = jest.fn();
+		test( 'should execute onContainerClick when clicked', () => {
+			const originalOnContainerClick = SelectDropdown.prototype.onContainerClick;
+			SelectDropdown.prototype.onContainerClick = jest.fn();
 
+			const fakeEvent = prepareFakeClickEvent( { } );
 			const dropdown = shallowRenderDropdown();
-			dropdown.find( '.select-dropdown__container' ).simulate( 'click' );
+			dropdown.find( '.select-dropdown__container' ).simulate( 'click', fakeEvent );
 
-			expect( SelectDropdown.prototype.toggleDropdown.mock.calls.length ).toBe( 1 );
-			SelectDropdown.prototype.toggleDropdown = originalToggleDropdown;
+			expect( SelectDropdown.prototype.onContainerClick.mock.calls.length ).toBe( 1 );
```

Comment:
We should check that the `toggleDropdown` method wasn't called to let tests catch any future regression. Also: we can add a new test to make sure the `toggelDropdown` method is called at some point. I guess we can do this by simulating a click on one of the parent divs.

Reply:
I hit a snag here, I'm not sure how to set this up correctly in jest – do you have experience with this, can you point me to an example? The problem is that the behaviour we want to test depends on the DOM state (to check where the click occurred).

It looks like there is [some support for mocking refs](https://reactjs.org/blog/2016/11/16/react-v15.4.0.html#mocking-refs-for-snapshot-testing), but it looks like it's focused more on snapshot testing. Maybe snapshot testing is a better fit here, since we're testing DOM changes – what do you think? 

Another option would be to factor out the DOM ref check, and unit test a method that takes a boolean `clickTargetIsSearchBox` parameter. I'm happy to do this if you think that's a good approach. 

I'm pretty new to testing in React and Jest (and snapshot testing!), so happy to take advice here :)

Reply:
As per enzyme's docs:  `ShallowWrapper.simulate` [won't simulate](http://airbnb.io/enzyme/docs/api/ShallowWrapper/simulate.html#common-gotchas) an actual DOM event, so we need to use their `Full DOM rendering.simulate` [API](http://airbnb.io/enzyme/docs/api/ReactWrapper/simulate.html) to test what we want.